### PR TITLE
adding hero list

### PR DESCRIPTION
**Pull request by tuzwoj**
It now displays one page of heroes from SWAPI.

### CODE REVIEW

File: src/app/app.component.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
+*{
+    box-sizing: border-box;
+    font-family: 'Roboto', sans-serif;
+    // background-color: ;
```
Comment (raibait-vizlib) [inline src/app/app.component.scss:4]: dont leave comments, unless there is a good reason

File: src/app/app.component.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+*{
+    box-sizing: border-box;
+    font-family: 'Roboto', sans-serif;
```
Comment (raibait-vizlib) [inline src/app/app.component.scss:3]: global styles place in style.scss

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,534 +1,4 @@
...
-<!-- * * * * * * * * * * End of Placeholder * * * * * * * * * * * -->
-<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
-
-
-
-<router-outlet></router-outlet>
\ No newline at end of file
+<div class="container">
+  <app-logo></app-logo>
+  <app-list></app-list>
```
Comment (raibait-vizlib) [inline src/app/app.component.html:3]: the components name is too generic , probably hero-list would be better , as there may be different  lists

File: src/app/app.module.ts [hunk 4]
```diff
@@ -1,16 +1,22 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
```
Comment (raibait-vizlib) [inline src/app/app.module.ts:4]: is that empty space on purpose ? :)

File: src/app/hero.service.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class HeroService {
+  // private heroesUrl = 'https://superheroapi.com/api/4025045517609315/search/batman';
+  private heroesUrl = 'https://swapi.dev/api/people/';
+
+  constructor(private http: HttpClient) { }
+
+  getHeroes(): Observable<any> {
```
Comment (raibait-vizlib) [inline src/app/hero.service.ts:15]: lacks model

File: src/app/hero.service.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  getHeroes(): Observable<any> { 
+    // const headerDict = {
+    //   'x-rapidapi-key': 'b0e4e1e864msh90944bd9a785585p1b5088jsn514fd2a5d106',
+    //   'x-rapidapi-host': 'free-nba.p.rapidapi.com'
+    // }
+    // const requestOptions = {                                                                                                                                                                                 
+    //   headers: new HttpHeaders(headerDict), 
+    // };
+    return this.http.get<any>(this.heroesUrl);
```
Comment (raibait-vizlib) [inline src/app/hero.service.ts:23]: lacks model

File: src/app/list/list.component.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<div class="app-list-container">
+    <div class="app-list">
+        <div class="app-list-header">
+            <span class="app-list-header-name">Heroes</span>
+
```
Comment (raibait-vizlib) [inline src/app/list/list.component.html:5]: try to make meaningful  spaces in a html template , this one seems random :D

File: src/app/list/list.component.scss [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+.app-list-header{
+    width:100%;
+    height: 50px;    
+    // background-image: linear-gradient(90deg, rgba(161,79,215,1) 0%, rgba(253,29,29,1) 50%, rgba(252,176,69,1) 100%);
+    border: 1px solid rgba(0,0,0,0.4);
+    background: rgba(0,0,0,0.1);
+    border-radius: 6px 6px 0px 0px;
+    color: rgba(0,0,0,0.8);
+    font-size: 1.5rem;
+
```
Comment (raibait-vizlib) [inline src/app/list/list.component.scss:25]: random space

File: src/app/list/list.component.scss [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    background-color: rgba(252, 176, 69, 0.4);
+    cursor: pointer;
+}
+
+.app-list-content-element>*{
+    height: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    font-size: 1.25rem;
```
Comment (raibait-vizlib) [inline src/app/list/list.component.scss:68]: in scss you can make a nicer syntax using &  i.e. 
.app-list-content-element{
  &:hover {
  
  }
  
  &>* {
  
  }
}

**Review by raibait-vizlib — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 10]
```diff
@@ -6,6 +6,9 @@
   <base href="/">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
[the remark this answers is not in the document]
Comment (tuzwoj) [reply src/index.html:10]: I use Roboto from Google fonts and it's added by them by default, when importing fonts.

File: src/app/app.module.ts (hunk 4 above)
Comment (tuzwoj) [reply src/app/app.module.ts:4]: Is it bad idea to visually separate external modules imports (such as @angular) from internal imports (I mean importing from files which are created within the project)?

Comment (raibait-vizlib) [reply src/app/app.module.ts:4]: its good idea :) i was just wonder if that was an accident
lets follow this pattern in this project,
 we dont use it in a project tho

**Review by raibait-vizlib — APPROVED** [review]

**Pull request merged**